package Sikozu.Messages;
// Header:
// uint8 Channel
// uint8 Command
// uint8 SID[18]
// uint8 NID[20]
// -> header size = 40 bytes


message Contact {
  required bytes nid= 1;
  optional uint32 ipv4 = 2;
  optional bytes ipv6 = 3;
  required uint32 port = 4;
}

message PingRequest {
}

message PingResponse {
}

message FindNodeRequest {
  required bytes nid = 1;
  optional string service = 2;
}

message FindNodeResponse {
  repeated Contact contacts = 1;
}

message GetServicesRequest {
}

message GetServicesResponse {
	message Result {
	  required uint32 channel = 1;
	  required string name = 2;
	}
  repeated Result services = 1;
}

message GetChannelRequest {
  required string name = 1;
}

message GetChannelResponse {
  required uint32 channel = 1;
}

message AnnounceServiceRequest {
	required bytes nid = 1;
	repeated string service = 2;
}

message AnnounceServiceResponse {
}

